ui/raidboss/data/06-ew/dungeon/alzadaals_legacy.ts

```diff
@@ -17,12 +46,124 @@ const triggerSet: TriggerSet<Data> = {
       response: Responses.aoe(),
     },
     {
-      id: 'Alzadaal Tentacle Dig',
-      type: 'StartsUsing',
-      netRegex: { id: ['6F55', '6559'], source: 'Ambujam', capture: false },
-      infoText: (_data, _matches, output) => output.text!(),
+      id: 'Alzadaal Tentacle Dig Collect',
+      type: 'MapEffect',
+      netRegex: { flags: tentacleFlags },
+      run: (data, matches) => data.tentacleEffects.push(matches),
+    },
+    {
+      id: 'Alzadaal Tentacle Dig Single',
+      type: 'Ability',
+      netRegex: { id: '6F55', source: 'Ambujam', capture: false },
+      delaySeconds: 2.3,
+      infoText: (data, _matches, output) => {
+        // only one tentacle (scarlet) for the first use of Tentacle Dig
+        if (data.tentacleEffects.length !== 1)
+          return output.default!();
+
+        const safeMap: { [key in Intercard]: Intercard } = {
+          NW: 'SE',
+          NE: 'SW',
+          SE: 'NW',
+          SW: 'NE',
+        };
+
+        const tentacleFlag = data.tentacleEffects[0]!.flags;
+        if (tentacleFlag === undefined || scarletTentacleLocations[tentacleFlag] === undefined)
+          return output.default!();
+        const tentacleLocation = scarletTentacleLocations[tentacleFlag]!;
```

Comment:
In general, I want to try to avoid non-null assertions (`!`) because they could be incorrect.  You can usually get around the need for this by storing things in separate variables so that TypeScript can narrow the type for just that variable.  It can't do type narrowing on a specific index of an object.

The one exception is for `output` since TypeScript doesn't know about proxies, and it would be a giant pain to undefined-check all of those.

```suggestion
        const tentacleFlag = data.tentacleEffects[0]?.flags;
        if (tentacleFlag === undefined)
          return output.default!();
        const tentacleLocation = scarletTentacleLocations[tentacleFlag];
        if (tentacleLocation === undefined)
          return output.default!();
```